logr.go

```diff
@@ -5,20 +5,21 @@
 // This design derives from Dave Cheney's blog:
 //     http://dave.cheney.net/2015/11/05/lets-talk-about-logging
 //
+// The logging specifically makes it non-trivial to use format strings, to encourage
+// attaching structured information instead of unstructured format strings.
+//
 // This is a BETA grade API.  Until there is a significant 2nd implementation,
 // I don't really know how it will change.
 package logr
 
-// TODO: consider structured logging, a la uber-go/zap
+// TODO: consider adding back in format strings if they're really needed
+// TODO: consider other bits of zap/zapcore functionality like ObjectMarshaller (for arbitrary objects)
 // TODO: consider other bits of glog functionality like Flush, InfoDepth, OutputStats
 
 // InfoLogger represents the ability to log non-error messages.
 type InfoLogger interface {
-	// Info logs a non-error message.  This is behaviorally akin to fmt.Print.
-	Info(args ...interface{})
-
-	// Infof logs a formatted non-error message.
-	Infof(format string, args ...interface{})
+	// Info logs a non-error message with the given key/value pairs as context.
```

Comment:
We should document the expected contents of the variadic argument.  e.g. is it:

```
log.Info("Oh no!", "condition", condition)
```
...and what the behavior is if the list is not evenly numbered

or
 
```
log.Info("Oh no!", log.KV("condition", condition))
```

Reply:
I'll document better.  It's `log.Info(msg, k1, v1, k2, v2, ...)`, matching the sugared logging from zap.  Importing zap's typechecked logging components seemed cumbersome.  `log.KV` feels a bit unwieldy, too.